lib/ansible/modules/windows/win_unzip.ps1

```diff
@@ -19,69 +19,78 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
+$ErrorActionPreference = "Stop"
 
-$params = Parse-Args $args;
+$pcx_extensions = @('.bz2', '.gz', '.msu', '.tar', '.zip')
 
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$src = Get-AnsibleParam -obj $params -name "src" -type "path" -failifempty $true
+$dest = Get-AnsibleParam -obj $params -name "dest" -type "path" -failifempty $true
+$creates = Get-AnsibleParam -obj $params -name "creates" -type "path"
+$recurse = Get-AnsibleParam -obj $params -name "recurse" -type "bool" -default $false
+$rm = Get-AnsibleParam -obj $params -name "rm" -type "bool" -default $false
+
+# Fixes a fail error message (when the task actually succeeds) for a
+# "Convert-ToJson: The converted JSON string is in bad format"
+# This happens when JSON is parsing a string that ends with a "\",
+# which is possible when specifying a directory to download to.
+# This catches that possible error, before assigning the JSON $result
 $result = @{
-    win_unzip = @{}
```

Comment:
Agreed, it seems like some older modules have it and either people just continued the tradition. Maybe I should just remove the older parameters and return values for the IIS stuff I've got going. I'll wait for comments on those as it would make things a lot simpler.